src/test/java/ArrayIndexedBagTest.java

```diff
@@ -7,406 +12,513 @@
+            }
+
+            @Test
+            void remove_negativeIndex_throwsIndexOutOfBoundsException() {
+                assertThrows(IndexOutOfBoundsException.class, () -> classUnderTest.remove(-1));
+            }
+
+            @Test
+            void remove_tooLargeIndex_throwsIndexOutOfBoundsException() {
+                assertThrows(IndexOutOfBoundsException.class, () -> classUnderTest.remove(1));
+            }
+
+            @Test
+            void remove_existingElement_returnsTrue() {
+                assertTrue(classUnderTest.remove(Integer.valueOf(10)));
+            }
+
+            @Test
+            void remove_existingElement_emptyCollection() {
+                classUnderTest.remove(Integer.valueOf(10));
+                assertEquals(new ArrayIndexedBag<Integer>(), classUnderTest);
+            }
+
+            @Test
+            void remove_nonexistentElement_throwsNoSuchElementException() {
+                assertThrows(NoSuchElementException.class, () -> classUnderTest.remove(Integer.valueOf(0)));
+            }
+
+            @Test
+            void indexOf_existingElement_returnsCorrectIndex() {
+                assertEquals(0, classUnderTest.indexOf(10));
+            }
+
+            @Test
+            void indexOf_nonexistentElement_throwsNoSuchElementException() {
+                assertThrows(NoSuchElementException.class, () -> classUnderTest.indexOf(0));
+            }
+
+            @Test
+            void contains_existingElement_returnsTrue() {
+                assertTrue(classUnderTest.contains(10));
+            }
+
+            @Test
+            void contains_nonexistentElement_returnsFalse() {
+                assertFalse(classUnderTest.contains(0));
+            }
+
+            @Test
+            void count_existingElement_returnsOne() {
+                assertEquals(1, classUnderTest.count(10));
+            }
+
+            @Test
+            void count_nonexistentElement_returnsZero() {
+                assertEquals(0, classUnderTest.count(0));
+            }
+
+            @Test
+            void isEmpty_singleton_returnsFalse() {
+                assertFalse(classUnderTest.isEmpty());
+            }
+
+            @Test
+            void size_singleton_returnsOne() {
+                assertEquals(1, classUnderTest.size());
+            }
+
+            @Test
+            void iterator_singleton_hasNextTrueOnce() {
+                Iterator<Integer> iterator = classUnderTest.iterator();
+                assertTrue(iterator.hasNext());
+                iterator.next();
+                assertFalse(iterator.hasNext());
+            }
+
+            @Test
+            void iterator_singleton_nextReturnsCorrectElements() {
+                Iterator<Integer> iterator = classUnderTest.iterator();
+                assertEquals(10, iterator.next());
+                assertThrows(NoSuchElementException.class, () -> iterator.next());
+            }
```

Comment:
Idk about these iterator tests. Not just the empty ones, but throughout. 

Noting here to bring attention to them.

Reply:
For iterator tests, I'd either make a separate test class for them, or just collect the iterator into a list and assert equality on the list. The latter is less helpful for finding actual bugs but it's much simpler to write as it works for any cardinality.